store/src/main/java/org/apache/rocketmq/store/queue/BatchConsumeQueue.java

```diff
@@ -710,6 +710,10 @@ public long getOffsetInQueueByTime(final long timestamp) {
         try {
             ByteBuffer byteBuffer = sbr.getByteBuffer();
             int left = targetMinOffset.getIndexPos(), right = targetMaxOffset.getIndexPos();
+            long maxQueueTimestamp = byteBuffer.getLong(right + MSG_STORE_TIME_OFFSET_INDEX);
+            if (timestamp >= maxQueueTimestamp) {
+                return byteBuffer.getLong(right + MSG_BASE_OFFSET_INDEX);
+            }
             int mid = binarySearchRight(byteBuffer, left, right, CQ_STORE_UNIT_SIZE, MSG_STORE_TIME_OFFSET_INDEX, timestamp);
             if (mid != -1) {
```

Comment:
if mid == -1, just the case timestamp > maxQueueTimestamp.

Reply:
Yes. Thanks for your reminder. If there return -1, it will get the min offset of this queue because of the logic the first picture shows. What's more, it's unnecessary to binary search if the target timestamp is already larger than the max timestamp in CQ.